### PR TITLE
CONSOLE-3761: (deps) replace `monaco-editor-core` with `monaco-editor` and bump version

### PR DESCRIPTION
**Pull request by logonoff**
<h2 id="feature-fix">CONSOLE Features and Fixes:</h2>

https://issues.redhat.com/browse/CONSOLE-3761

<h2 id="solution-description">Solution description</h2> 

Replace monaco-editor-core with monaco-editor to fix build errors and to satisfy previously invalid peer deps, but keep an older version until migration to newer webpack due to build errors from newer javascript syntax usage in newer versions of monaco-editor

### CODE REVIEW

**Comment by logonoff** [comment]
Sorry about the change, I just noticed a regression with the light theme after updating the package (the current line indicator was still in dark mode), this has now been fixed

File: frontend/packages/integration-tests-cypress/support/a11y.ts [hunk 1]
```diff
@@ -66,6 +66,7 @@ Cypress.Commands.add('testA11y', (target: string, selector?: string) => {
       { id: 'color-contrast', enabled: false }, // seem to be somewhat inaccurate and has difficulty always picking up the correct colors, tons of open issues for it on axe-core
       { id: 'focusable-content', enabled: false }, // recently updated and need to give the PF team time to fix issues before enabling
       { id: 'scrollable-region-focusable', enabled: false }, // recently updated and need to give the PF team time to fix issues before enabling
+      { id: 'aria-roles', enabled: false }, // rule is outdated and current installation of axe-core is no longer complaint with newest ariaRoles spec, causing false test failures (specifically with the role='code'). reenable when axe core is updated to at least 4.1.0
```
Comment (logonoff) [inline frontend/packages/integration-tests-cypress/support/a11y.ts:69]: note that an accessibility check is disabled due to an outdated implementation in our current version of `axe-core`, upgrading in turn is blocked by various accessibility issues in console that must be resolved

**Review by vikram-raj — COMMENTED** [review]
/lgtm

**Comment by sanketpathak** [comment]
Verified on a cluster with admin and non-admin users. I also tried the test run which is using yaml view. Runs green
Screenshot from 2024-10-04 21-14-09
Screenshot from 2024-10-04 22-06-39
[Screencast from 2024-10-04 22-12-45.mov](https://github.com/user-attachments/assets/d5f84511-2efc-424b-ad24-f418671d28aa)

/label qe-approved

**Comment by logonoff** [comment]
this PR does not impact docs or product experience

/label docs-approved
/label px-approved

**Pull request merged**